lib/practice_exercises.rb

```diff
@@ -1,13 +1,48 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(3n + n/2) == O(n)
+# Space Complexity: O(1)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  string = string.downcase                # .downcase TC = O(n)
+  string = string.delete "^a-z"           # .delete TC = O(n)
+
+  first = 0
+  last = string.length - 1                # .length TC = O(n)
+
+  until first >= last                     # loop TC = O(n/2) == O(n)
+    if string[first] != string[last]
+      return false
+    end
+    first += 1
+    last -= 1
+  end
+
+  return true
+
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2 + 2n) == O(n^2)???
+# Space Complexity: O(n)
 def longest_prefix(strings)
```

Comment:
👍 , Correct, this works in O(n^2) time complexity.  Later we'll see ways to bring this down to O(n).